diff --git a/Cargo.toml b/Cargo.toml
index 3aa40a6..babda04 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,9 +28,6 @@ edition = "2021"
 categories = ["command-line-utilities", "parsing"]
 keywords = ["pattern", "regex", "regexp"]
 
-[lib]
-crate-type = ["cdylib", "rlib"]
-
 [dependencies]
 atty = { version = "0.2.14", optional = true }
 clap = { version = "4.0.18", features = ["derive", "wrap_help"], optional = true }
@@ -43,7 +40,7 @@ unic-char-range = "0.9.0"
 unic-ucd-category = "0.9.0"
 unicode-segmentation = "1.10.0"
 
-[target.'cfg(target_family = "wasm")'.dependencies]
+[target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dependencies]
 wasm-bindgen = "0.2.81"
 
 [dev-dependencies]
@@ -57,7 +54,7 @@ predicates = "2.1.1"
 proptest = "1.0.0"
 tempfile = "3.3.0"
 
-[target.'cfg(target_family = "wasm")'.dev-dependencies]
+[target.'cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))'.dev-dependencies]
 wasm-bindgen-test = "0.3.31"
 
 [features]
diff --git a/src/lib.rs b/src/lib.rs
index 0658111..3a4e7a7 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -278,10 +278,10 @@ mod regexp;
 mod substring;
 mod unicode_tables;
 
-#[cfg(target_family = "wasm")]
+#[cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))]
 mod wasm;
 
 pub use builder::RegExpBuilder;
 
-#[cfg(target_family = "wasm")]
+#[cfg(all(target_arch = "wasm32", not(any(target_os = "emscripten", target_os = "wasi"))))]
 pub use wasm::RegExpBuilder as WasmRegExpBuilder;
